release-calendar/src/client/components/ReleaseCard.tsx

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../stylesheets/releaseCard.scss';
+import * as React from 'react';
+import {ApiService} from '../api-service';
+import {Channel} from '../../types';
+import {EventApi} from '@fullcalendar/core';
+import {ReleaseDates} from '../models/view-models';
+import {channelTitles} from './ChannelTable';
+import moment from 'moment-timezone';
+
+export interface ReleaseCardProps {
+  eventApi: EventApi;
+}
+
+interface ReleaseCardState {
+  releaseDates: ReleaseDates;
+}
+
+export class ReleaseCard extends React.Component<
+  ReleaseCardProps,
+  ReleaseCardState
+> {
+  private apiService: ApiService;
+
+  constructor(props: Readonly<ReleaseCardProps>) {
+    super(props);
+    this.state = {
+      releaseDates: null,
+    };
+    this.apiService = new ApiService();
+  }
+  async componentDidMount(): Promise<void> {
+    const releaseDates: ReleaseDates = await this.apiService.getReleaseDates(
+      this.props.eventApi.title,
+    );
+    this.setState({releaseDates});
+  }
+
+  history = [
+    {
+      channel: Channel.NIGHTLY,
+      emoji: '🌙',
+      emojiName: 'moon',
+    },
+    {
+      channel: Channel.OPT_IN_EXPERIMENTAL,
+      emoji: '✋',
+      emojiName: 'hand',
+    },
+    {
+      channel: Channel.PERCENT_EXPERIMENTAL,
+      emoji: '🧪',
+      emojiName: 'experiment',
+    },
+    {
+      channel: Channel.STABLE,
+      emoji: '🏠',
+      emojiName: 'house',
+    },
+    {
+      channel: Channel.LTS,
+      emoji: '🏙️',
+      emojiName: 'city',
+    },
+  ];
+
+  getTitle(search: Channel): string {
```

Comment:
What if `history` was a dict with Channel as the key? Could be a lot cleaner here.

Reply:
The best solution I can think to debug the channels with "-" not being accessible within the dictionary is to create an interface where the key is typed as a string. This will push the enum into a string under the hood, but it allows for normal dictionary sets and gets. If you are okay with here, then I could also make the channelsTitles a dictionary, which is currently a {channel: Channel, title: string}[]. We had bad it this because of the same weird "-" enum behavior. If I switched it, then I'd again have direct access to the titles I want instead of doing the find() loop I do now.

Reply:
Can we revisit the dict idea? Hyphens should not be a problem. For example
```
const channelMap = {
    'lts': 'lts',
    'stable': 'stable',
    'opt-in-beta': 'opt in beta',
    'perc-beta': '1% beta',
};

console.log(channels['opt-in-beta']);
// opt in beta

console.log(channels['perc-beta']);
// 1% beta
```

Reply:
Making channelsTitles a dictionary is a great idea :)

Reply:
Also I totally misread your comment! My "revisit the dict idea" was not helpful haha. Nice workaround.